File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/forecasts/DayPeriod.java
package io.github.tobyhs.weatherweight.api.accuweather.forecasts;

import com.google.auto.value.AutoValue;
import com.google.gson.annotations.SerializedName;
import me.tatarka.gsonvalue.annotations.GsonBuilder;

/**
 * Forecast details about a period of the day (day or night)
 */
@AutoValue
public abstract class DayPeriod {
    /**
     * @return phrase description of the icon corresponding to the forecast
     */
    @SerializedName("IconPhrase")
    public abstract String getIconPhrase();

    @GsonBuilder
    public static Builder builder() {
        return new AutoValue_DayPeriod.Builder();
    }

    @AutoValue.Builder
    public static abstract class Builder {
        public abstract Builder setIconPhrase(String iconPhrase);
        public abstract DayPeriod build();
    }
}

File: app/src/main/java/io/github/tobyhs/weatherweight/api/accuweather/forecasts/DayPeriod.kt
package io.github.tobyhs.weatherweight.api.accuweather.forecasts

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Forecast details about a period of the day (day or night)
 *
 * @property iconPhrase phrase description of the icon corresponding to the forecast
 * @property precipitationProbability percent representing the probability of precipitation
 */
@JsonClass(generateAdapter = true)
data class DayPeriod(
    @Json(name = "IconPhrase") val iconPhrase: String,
    @Json(name = "PrecipitationProbability") val precipitationProbability: Int,
)
